ModRemote.rbxs

```diff
@@ -1,423 +1,391 @@
 end
 
---- Creates a function
--- @param string name - the name of the function.
-function remote:CreateFunction(name)
-		
-	
-	if (not server) then
-		warn("[ModRemote] CreateFunction should be used by the server."); end
-	
-	return remote.internal:CreateFunction(name);
-end
+function remoteFunction:ResetClientCache()
 
+	assert(not server, "ResetClientCache must be used on the client.")
+	local instance = self.Instance
 
---======== REMOTE FUNCTION ===========
-remote.FuncCache = {};
+	if instance:FindFirstChild("ClientCache") then
+		FuncCache[instance:GetFullName()] = {Expires = 0, Value = nil}
+	else
+		warn(instance:GetFullName() .. " does not have a cache.")
+	end		
+end
 
-do -- [[REMOTE FUNCTION OBJECT METHODS ]]
-	local remFunc = remote.func;
+function remoteFunction:CallServer(...)
+	assert(not server, "[ModRemote] CallServer should be called from the client side.")
 
-	function remFunc:CallPlayer(player, ...) 
-		
-		assert(server, "[ModRemote] CallPlayer should be called from the server side."); 	
-		
-		local args = {...};
-		local attempt, err = pcall(function()
-			return self.Instance:InvokeClient(player, unpack(args));
-		end);
-		
-		if (not attempt) then
-			warn("[ModRemote] CallPlayer - Failed to recieve response from " .. player.Name);
-			return nil;
-		end	
-	end
-	
-	function remFunc:CallServerIntl(...) 
-		assert(client, "[ModRemote] CallServer should be called from the client side."); 	
-		return self.Instance:InvokeServer(...);
-	end
-	
-	function remFunc:Callback(func)
-		if (server) then
-			self.Instance.OnServerInvoke = func;
-		else
-			self.Instance.OnClientInvoke = func;
-		end	
-	end
-	
-	function remFunc:GetInstance()
-		return self.Instance;
-	end
-	
-	function remFunc:Destroy()
-		self.Instance:Destroy();
-	end
-	
-	function remFunc:SetClientCache(seconds, useAction)
-		
-		
-		seconds = seconds or 10;
-		assert(server, "SetClientCache must be called on the server.");
-		local instance = self:GetInstance();
-		
-		if (seconds == false or seconds < 1) then
-			local cache = instance:FindFirstChild("ClientCache");
-			if (cache) then
-				cache:Destroy();
-			end
-		else
-			local cache = instance:FindFirstChild("ClientCache") or Instance.new("IntValue", instance);
-			cache.Name = "ClientCache";
-			cache.Value = seconds;
-		end
-		
-		if (useAction) then
-			local cache = instance:FindFirstChild("UseActionCaching") or Instance.new("BoolValue", instance);
-			cache.Name = "UseActionCaching";
-			cache.Value = true;
-		else
-			local cache = instance:FindFirstChild("UseActionCaching");
-			if (cache) then
-				cache:Destroy();
-			end			
-		end
-		
-	end
-	
-	function remFunc:ResetClientCache()
-				
-		
-		assert(client, "ResetClientCache must be used on the client.");
+	local instance = self.Instance
+	local clientCache = instance:FindFirstChild("ClientCache")
+
+	if clientCache then
+		local cacheName = instance:GetFullName() .. (instance:FindFirstChild("UseActionCaching") and tostring({...}[1]) or "")
```

Comment:
This line needs brackets around the arg table to work, otherwise it errors.

```lua
local cacheName = instance:GetFullName() .. (instance:FindFirstChild("UseActionCaching") and tostring(({...})[1]) or "")
```